src/main/java/io/odpf/depot/bigquery/json/BigqueryJsonUpdateListener.java

```diff
@@ -1,24 +1,98 @@
 package io.odpf.depot.bigquery.json;
 
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.Schema;
+import io.odpf.depot.bigquery.handler.BigQueryClient;
 import io.odpf.depot.bigquery.handler.MessageRecordConverter;
 import io.odpf.depot.bigquery.handler.MessageRecordConverterCache;
+import io.odpf.depot.bigquery.proto.BigqueryFields;
+import io.odpf.depot.common.TupleString;
 import io.odpf.depot.config.BigQuerySinkConfig;
 import io.odpf.depot.message.OdpfMessageParser;
 import io.odpf.depot.stencil.OdpfStencilUpdateListener;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 public class BigqueryJsonUpdateListener extends OdpfStencilUpdateListener {
     private final MessageRecordConverterCache converterCache;
     private BigQuerySinkConfig config;
+    private final BigQueryClient bigQueryClient;
 
-    public BigqueryJsonUpdateListener(BigQuerySinkConfig config, MessageRecordConverterCache converterCache) {
+    public BigqueryJsonUpdateListener(BigQuerySinkConfig config, MessageRecordConverterCache converterCache, BigQueryClient bigQueryClient) {
         this.converterCache = converterCache;
         this.config = config;
+        this.bigQueryClient = bigQueryClient;
+        if (!config.getSinkConnectorSchemaJsonDynamicSchemaEnable()) {
+            throw new UnsupportedOperationException("currently only schema inferred from incoming data is supported, stencil schema support for json will be added in future");
+        }
     }
 
     @Override
     public void updateSchema() {
         OdpfMessageParser parser = getOdpfMessageParser();
         MessageRecordConverter messageRecordConverter = new MessageRecordConverter(parser, config, null);
         converterCache.setMessageRecordConverter(messageRecordConverter);
+        Schema existingTableSchema = bigQueryClient.getSchema();
+        FieldList existingTableFields = existingTableSchema.getFields();
+        List<TupleString> defaultColumns = config.getSinkBigquerySchemaJsonOutputDefaultColumns();
+        Set<String> defaultColumnNames = defaultColumns
```

Comment:
can we do lambda 
`
        Set<String> defaultColumnNames = defaultColumns
                .stream()
                .map(TupleString::getFirst)
                .collect(Collectors.toSet());
        HashSet<Field> fieldsToBeUpdated = defaultColumns
                .stream()
                .map(this::getField)
                .collect(Collectors.toCollection(HashSet::new));
`